src/sysext/sysext.c

```diff
@@ -262,9 +277,38 @@ static int unmerge_hierarchy(
                 if (r == 0)
                         break;
 
+                r = read_one_line_file(work_dir_info_file, &base64_work_dir);
+                if (r < 0) {
+                        if (r != -ENOENT)
+                                return log_error_errno(r, "Failed to read '%s': %m", work_dir_info_file);
+                }
+
+                if (base64_work_dir) {
```

Comment:
just make this an "else"? it will only be non-NULL if read_one_line_file() succeeded after alll

Reply:
Done.